backend/census_historical_migration/workbooklib/federal_awards.py

```diff
@@ -88,6 +88,32 @@
 ]
 
 
+def xform_missing_major_program(audits):
+    """Default missing major program by extrapolating from audit report type."""
+    change_records = []
+    is_empty_major_program_found = False
+
+    for audit in audits:
+        major_program = string_to_string(audit.MAJORPROGRAM)
+        if not major_program:
+            new_value = "Y" if string_to_string(audit.TYPEREPORT_MP) else "N"
+
+            track_transformations(
```

Comment:
We want to call track_transformations regardless of `if not major_program` condition. However, we want save the record (i.e., `InspectionRecord.append_federal_awards_changes(change_records)`) only when `is_empty_major_program_found` is true. @phildominguez-gsa